learn.md

```diff
@@ -1,50 +1,262 @@
+
+## Tools:
+
+### Required Tools to start for free today
+
+- [Ghidra](https://ghidra-sre.org/)
+  - Use this to decompile binaries and executables
+- [GDB](gnu.org/software/gdb/)
+  - Useful for dynamic analysis
+- [Hex Editor](https://hexed.it/)
+  - Useful for editing binaries
+- [**GOOGLE**](https://www.google.com/)
+  - You will most likely know nothing about the challenge. That's one of the main componennts of reverse engineering. **Learn how to google or this category won't be fun for you.**
+
+### More advanced tool classifications and usages
+
+#### Static Analysis Tool
+- Ghidra
+- Radare2
+- Binary Ninja
+- Ida
+#### Dynamic Analysis Tool
+- GDB
+- Binary Ninja
+- Ida
+#### Advanced Automatic solving tools
+- [Angr](https://angr.io/)
+  - Does everything you will try to do but better
+  - Make a virtual environment in python or you will break your python
+#### Must have python libraries
+- [Pwntools](https://pypi.org/project/pwntools/)
+  - Gives better access to the binaries when running scripts
+- [Z3py Solver](https://pypi.org/project/z3-solver/)
+  - Used to check satisfiability of equations
+  - Useful for flag checker challenges
+#### C# specific challenges
+- [dnSpy](https://github.com/dnSpy/dnSpy)
+  - Decompiles C#
+  - Lets you edit the C# and when you save it will recompile for you
+#### APK specific challenges
+- [MobSF](https://mobsf.github.io/docs/#/)
+  - All in one APK tool 
+  - Extracts all APK data
+  - Decompiles all java
+  - Sets up emulation for dynamic analysis
+#### Compiled python challenges
+- [Uncompyle](https://pypi.org/project/uncompyle6/)
+    - Decompiles compiled python bytecode
+- [Pyinstxtractor](https://github.com/extremecoders-re/pyinstxtractor)
+  - Used for extracting compiled python from executables and binaries made with pyinstaller
+
+## How to Learn:
+
+### Practice, Practice, Practice. 
+- Reverse engineering is extremely time consuming, especially when you start, but as you practice you will start recognizing pattern and get faster each time.
+- Almost all good Reverse engineering challenges will be unique and will require that you will need to learn about something new, this skill improves as you practice.
+- You can practice lots of challenges [here](https://github.com/Kasimir123/CTFWriteUps). All the challenges are rated for difficulty and each challenge has the original files and a writeup.
+
+
+### GuidedHacking is an amazing forum with a lot of information about game hacking.
+- Always remember to only hack things you have permission to hack/ try to stay on single player games to stay safe.
+- Following their tutorials and hacking various games you get exposure to multiple architectures, learn reverse engineering tricks, and you will learn about how to get around anti debug code.
```

Comment:
Maybe move this to an "Other Topics" section? Game Hacking isn't very relevant to intro CTF reverse engineering